utilities/shared_disk.py

```
@@ -97,48 +102,107 @@ def _write_marker(ssh_conn: VMSSHConnection, file_path: str, content: str, vm_la
-    raise ValueError(f"No shared disk found for VM '{vm_name}' — all disks on same SCSI controller")
+    vm1_dest_name = resolve_destination_vm_name(vm1_config)
+    vm2_dest_name = resolve_destination_vm_name(vm2_config)
+
+    vm1_pvcs = _get_pvc_names_from_vm(ocp_admin_client, target_namespace, vm1_dest_name)
+    vm2_pvcs = _get_pvc_names_from_vm(ocp_admin_client, target_namespace, vm2_dest_name)
+
+    shared_pvcs = set(vm1_pvcs) & set(vm2_pvcs)
+    if not shared_pvcs:
+        raise ValueError(
+            f"No shared PVC between '{vm1_dest_name}' and '{vm2_dest_name}'. VM1 PVCs: {vm1_pvcs}, VM2 PVCs: {vm2_pvcs}"
+        )
```

Ruff (0.15.15)

[warning] Avoid specifying long messages outside the exception class

(TRY003)